### ISSUE TITLE
Vietnam phone numbers not validating for 2FA

### ISSUE THREAD

**Issue by pratikawari28** [Automattic/wp-calypso#25163]
Adding a phone number with an initial "0" or "8" after the Vietnam country code(+84) returns a validation error for 2FA.

**Steps to reproduce**
Starting at URL: https://wordpress.com/
Go to Security
Click Two-Step authentication > Get Started
Select “Vietnam” in Country code
Enter a phone number in the following format(country code-phone number) : +84-XXXXXXXXXX
Click “Verify via SMS” or “Verify via App”

**What I expected**
To be prompted to complete the verification process.

**What happened instead**
I saw the error, “That phone number does not appear to be valid”. It appears we do not support numbers starting with **0** and **8** for this country code.

**Browser / OS version**
Chrome
Firefox

**Screenshot / Video**
![](https://cldup.com/kx34XzWavH.png)

**Context / Source**
612695-hc
1190225-zen

#user-report
#manual-testing
p1527589855000520-slack-triage

**Comment by chad1008** [comment]
Similar issues: #23276, #21862 

From those, we found that [our current check for VN](https://github.com/Automattic/node-phone/blob/6be6549b03137f2cca01e202250bf2590750119e/lib/index.js#L1820) numbers only allows for them to begin with a 9 or a 1.

**Comment by KokkieH** [comment]
Another report from the [support forums](https://en.forums.wordpress.com/topic/my-phone-number-in-changed-but-wp-didnt-recognized-it/).

In this case the number has changed from `+84 169…` to `+84 39…` 

Based on https://en.wikipedia.org/wiki/Telephone_numbers_in_Vietnam#Mobile_telephone_numbers_[1]_[2] Vietnam converted all 11-digit mobile phone numbers to 10-digit instead, and it appears this has now gone into effect.

**Comment by KokkieH** [comment]
Based on [this news article](http://m.sggpnews.org.vn/science_technology/11digit-mobile-phone-numbers-to-be-changed-to-10digits-75203.html) numbers can now start with 3, 5, 7 and 8, in addition to 9, with the implementation happening from now through July next year. Looking at the Wikipedia page it appears 6 might also be used and 1 has fallen away completely.

**Comment by htdat** [comment]
Hi, I am from Vietnam. 

I confirm the change like what @KokkieH mentioned above. If we need any info, I can provide more info. 

Telecom providers are urging people to update numbers across services and 2FA with the following general timeline: 

* Between the time Sep 15 - Oct 15, 2018: people can use both 10 and 11 digit numbers. 
* After that, people can use just 11-digit numbers.

**Comment by jeremypaavola** [comment]
Report from a user of running into validation issues from Vietnam

3950579-hc
1415784-zen

**Comment by KokkieH** [comment]
Looks like there's ongoing internal discussion in p4TIVU-98g-p2 on how to handle these.

**Comment by lutuyen** [comment]
> Hi, I am from Vietnam.
> 
> I confirm the change like what @KokkieH mentioned above. If we need any confirmation, I can provide more or verify.
> 
> Telecom providers are urging people to update numbers across services and 2FA with the following general timeline:
> 
> * Between the time Sep 15 - Oct 15, 2018: people can use both 10 and 11 digit numbers.
> * After that, people can use just 11-digit numbers.

Just want to correct your mistake, after that Vietnamese people can only use 10-digit numbers!

**Comment by lutuyen** [comment]
> Based on [this news article](http://m.sggpnews.org.vn/science_technology/11digit-mobile-phone-numbers-to-be-changed-to-10digits-75203.html) numbers can now start with 3, 5, 7 and 8, in addition to 9, with the implementation happening from now through July next year. Looking at the Wikipedia page it appears 6 might also be used and 1 has fallen away completely.

Correct. Please update Wordpress system asap so that users can use Two-step Authentication properly. Thanks.

**Comment by htdat** [comment]
> Just want to correct your mistake, after that Vietnamese people can only use 10-digit numbers!

Thanks! Updated.

**Comment by lutuyen** [comment]
How's it going, guys? We need Two-Step Authentication updated for numbers starting with 7, 8, etc., to be able to be verified, please.

**Comment by htdat** [comment]
@lutuyen: Thanks for following up. In the meantime, I suggest these approaches: 

1. [Use Google Authenticator app](https://en.support.wordpress.com/security/two-step-authentication/#setup-with-google-authenticator).
2. [Create and save your backup codes](https://en.support.wordpress.com/security/two-step-authentication/#backup-codes). 

Our team is working on it. 

----- 

From [this article](https://news.zing.vn/dau-so-rut-gon-sim-11-so-viettel-nhan-03-mobifone-nhan-07-post846698.html) (not a goverment source but it can be trusted), the plan looks like this: 

### General plan: 

* Between the time Sep 15 - Nov 15, 2018: people can use both 10 and 11 digit numbers.
* After that, people can use just 10-digit numbers.

### Exact changes: 

|||||
|--- |--- |--- |--- |
|Old Prefix Numbers|New Prefix Numbers|Telcom Provider|Conversion starting date (dd/m)|
|0169.66|039.66|Viettel|15/9|
|0169|039|Viettel|17/9|
|0168|038|Viettel|19/9|
|0167|037|Viettel|23/9|
|0166|036|Viettel|25/9|
|0165|035|Viettel|27/9|
|0164|034|Viettel|3/10|
|0163|033|Viettel|5/10|
|0162|032|Viettel|7/10|
|0120|070|MobiFone|15/9|
|0121|079|MobiFone|24/9|
|0122|077|MobiFone|27/9|
|0126|076|MobiFone|4/10|
|0128|078|MobiFone|7/10|
|0124|084|VinaPhone|15/9|
|0127|081|VinaPhone|18/9|
|0129|082|VinaPhone|21/9|
|0123|083|VinaPhone|24/9|
|0125|085|VinaPhone|27/9|
|0186|056|Vietnamobile|15/9|
|0188|058|Vietnamobile|15/9|
|0199.2|059.2|Gtel|15/9|
|0199.3|059.3|Gtel|17/9|
|0199.8|059.8|Gtel|19/9|
|0199.9|059.9|Gtel|21/9|

**Comment by lutuyen** [comment]
Thanks. But I raised my the issue about 5 months ago. I have been using a number starting with 089 and the system never accepted it. The issue has still remained open up to this time.
One question, does it take that long to really solve such a small problem? I remember I once managed a web and software development company and my developers only take a day to have such updated!

**Comment by lutuyen** [comment]
Please don't forget the 10-digit number starting with 089. This is not on your list since it's already a 10-digit number and there's no need to convert it. You can check it at < https://en.wikipedia.org/wiki/Telephone_numbers_in_Vietnam#Mobile_telephone_numbers_%5B1%5D_%5B2%5D >
This number has already existed for about one and half years but Wordpress system has ignored it! that's why I've got verification issue with your system.

**Comment by grvrulz** [comment]
Another one in 1482922-zen.

**Comment by joendotcom** [comment]
1498768-zen

**Comment by kraftbj** [comment]
We use a forked version of the phone npm module which is out of date ( https://github.com/Automattic/node-phone/blob/master/lib/index.js#L1820 ) compared to the original ( https://github.com/AfterShip/phone/blob/master/lib/iso3166Data.js#L1812 ).

We originally forked it because of the slow pace of updates, but that seems to have changed course now.

**Comment by erricgunawan** [comment]
1566536-zen which comes from 1415784-zen (reported originally [here](https://github.com/Automattic/wp-calypso/issues/25163#issuecomment-422751497)) just asked an update about this.

**Comment by Janna-Pieler** [comment]
+1 at 1566027-zen

**Comment by synora** [comment]
@kraftbj is there any update on @Janna-Pieler 's comment above?
